src/poliastro/frames.py

```diff
@@ -220,6 +226,78 @@ class PlutoICRS(_PlanetaryICRS):
     body = Pluto
 
 
+def _make_rotation_matrix_from_reprs(start_representation, end_representation):
+    """
+    Return the matrix for the direct rotation from one representation to a second representation.
+    The representations need not be normalized first.
+    """
+    A = start_representation.to_cartesian()
+    B = end_representation.to_cartesian()
+    rotation_axis = A.cross(B)
+    rotation_angle = -np.arccos(
+        A.dot(B) / (A.norm() * B.norm())
+    )  # negation is required
+
+    # This line works around some input/output quirks of Astropy's rotation_matrix()
+    matrix = np.array(rotation_matrix(rotation_angle, rotation_axis.xyz.value.tolist()))
+    return matrix
+
+
+_EARTH_ORBIT_PERPEN_POINT_GCRS = UnitSphericalRepresentation(
+    lon=0 * u.deg, lat=66.5 * u.deg
+)
+
+_EARTH_DETILT_MATRIX = _make_rotation_matrix_from_reprs(
+    _EARTH_ORBIT_PERPEN_POINT_GCRS, CartesianRepresentation(0, 0, 1)
+)
+
+
+class GeocentricSolarEcliptic(BaseEclipticFrame):
+    """
+    This system has its X axis towards the Sun and its Z axis perpendicular to
+    the plane of the Earth's orbit around the Sun (positive North). This system
+    is fixed with respect to the Earth-Sun line. It is convenient for specifying
+    magnetospheric boundaries. It has also been widely adopted as the system for
+    representing vector quantities in space physics databases.
+
+    """
+
+    obstime = TimeAttribute(default=DEFAULT_OBSTIME)
+
+
+@frame_transform_graph.transform(DynamicMatrixTransform, GCRS, GeocentricSolarEcliptic)
+def gcrs_to_geosolarecliptic(gcrs_coo, to_frame):
+
+    if to_frame.obstime is None:
+        raise ValueError(
+            "To perform this transformation the coordinate"
+            " Frame needs an obstime Attribute"
+        )
+
+    sun_pos_icrs = get_body_barycentric("sun", to_frame.obstime)
```

Comment:
And would changing the frame of this matter? As we are finding the difference vector